### ISSUE TITLE
Background Clip shorthand breaking IE8

### ISSUE THREAD

**Issue by jaetea** [jakubpawlowicz/clean-css#577]
No matter what flag I set clean-css adding the background clip property to the background shorthand.

This completely break in IE8.

background:url(../../images/site_chrome_bg.png) repeat-x padding-box #6295b9;

padding-box need to stay in it's own property to work in IE8. background-clip: padding-box;

**Comment by jakubpawlowicz** [comment]
Is it also the thing in IE8 mode?

**Comment by jaetea** [comment]
Yes does this in IE8 compatibility mode too..which is strange.

**Comment by jakubpawlowicz** [comment]
Well, need to give it a try - we do similar thing for `background-size` already but I thought `background-clip` is fine in IE8.

**Comment by jaetea** [comment]
Just run this through clean-css

.portlet header {
  background: #6295b9 url('../../images/site_chrome_bg.png') repeat-x;
  text-align: left;
  border: 1px solid #cccccc;
  border-bottom: 0;
  font-weight: bold;
  padding: 4px 6px;
  margin: 0;
  -moz-border-radius-topright: 4px;
  -moz-border-radius-bottomright: 0;
  -moz-border-radius-bottomleft: 0;
  -moz-border-radius-topleft: 4px;
  border-top-right-radius: 4px;
  border-bottom-right-radius: 0;
  border-bottom-left-radius: 0;
  border-top-left-radius: 4px;
  -moz-background-clip: padding;
  background-clip: padding-box;
}

**Comment by jakubpawlowicz** [comment]
:+1: thx - need to test it in IE too - in the meantime you can disable it entirely using `--skip-shorthand-compacting`.

**Comment by jaetea** [comment]
So that works. Except now the space after the closing URL bracket isn't being written.
cleancss --compatibility *,+properties.spaceAfterClosingBrace --skip-shorthand-compacting -o models.css models.css

**Comment by jakubpawlowicz** [comment]
Can you try with `cleancss --compatibility ie8 --skip-shorthand-compacting -o models.css models.css`?

**Comment by jakubpawlowicz** [comment]
This is the output I get:

```css
portlet header{background:url(../../images/site_chrome_bg.png) repeat-x #6295b9;text-align:left;border:1px solid #ccc;border-bottom:0;font-weight:700;padding:4px 6px;margin:0;-moz-border-radius-topright:4px;-moz-border-radius-bottomright:0;-moz-border-radius-bottomleft:0;-moz-border-radius-topleft:4px;border-top-right-radius:4px;border-bottom-right-radius:0;border-bottom-left-radius:0;border-top-left-radius:4px;-moz-background-clip:padding;background-clip:padding-box}
```

**Comment by jaetea** [comment]
Perfect. Thanks so  much!

**Comment by jakubpawlowicz** [comment]
:+1: I'll keep the issue open though to investigate `background-clip` on IE8.

**Comment by jakubpawlowicz** [comment]
@jaetea it's fixed in master now, will be a part of upcoming 3.3 release.

As a side note it also affects `background-origin` merging which will be addressed in #579.
